### DISCUSSION TITLE
Segmented Control replacement in Paste?

### DISCUSSION THREAD

**Discussion by a-christensen** [twilio-labs/paste#903] [Q&A] [answered]
Hey crew!

**Context:** SendGrid is making a analytics product like Messaging Insights in Console. I want to keep as much parity between the two analytics interfaces as possible.

**Problem:** In Messaging Insights, we used the "Segmented Control" console component to change the visualization type (line chart, bar chart, pie chart, table, etc.). See an example in the top right corner of the graph content [here.](https://twilio.invisionapp.com/share/K2Y3MJ2NXG3#/411711549_Delivery_-_Errors-Delivery_Status) I want to try to do something similar in SendGrid but am unsure what Paste component maps to the "Segmented Control." If there isn't a direct replacement at the moment, I am happy to use whatever else is recommended in Paste.

**Question:** What component do you recommend I use for toggling between visualization types in SendGrid's Insights product (instead of the "Segmented Control")?

**Comment by corinne-nagel** [comment, marked as answer]
Hi @a-christensen , currently the button group is scheduled for "Release -2021.04.20 TBD" so it's a ways away at the moment: https://paste.twilio.design/roadmap. When were you thinking you might need it?

Some initial thoughts: 
- you could compose your own custom composition potentially using buttons as a design reference if you decide to go this route
- You could use a combobox/drop down to allow the user to choose an option from a list
- Since this is a selection of one thing among others, you could also consider using a radio button. 

The key is to show the pressed/selected state.

I'll also caveat all this by saying please test whatever direction you go with users to make sure you're designing the best option :) 

(And if you decide to do a custom composition that you think would map well to the Button Group component/have time to do such, please consider contributing!)

**Reply by a-christensen** [reply]
Thanks, Corinne! The engineers will probably start building this in December 2020 so I don't think the 2021 date will work for us. 

Thanks for laying out some options. This will definitely go through usability testing so no worries on that front. I'll play around with those options you suggested and keep you in the loop if I do anything custom. Thanks again!

**Comment by corinne-nagel** [comment]
Awesome, thanks for checking in!
